test/libsolidity/SolidityNatspecJSON.cpp

```diff
@@ -485,6 +485,309 @@ BOOST_AUTO_TEST_CASE(event)
 	checkNatspec(sourceCode, "ERC20", userDoc, true);
 }
 
+BOOST_AUTO_TEST_CASE(emit_same_signature_event_library_contract)
+{
+	char const* sourceCode = R"(
+		library L {
+		    /// @notice This event is defined in Library L
+		    /// @dev This should not appear in Contract C dev doc
+		    event SameSignatureEvent(uint16);
+		    /// @notice This event is defined in Library L
+		    /// @dev This should not appear in Contract C dev doc
+		    event LibraryEvent(uint32);
+		}
+		contract C {
+		    /// @notice This event is defined in Contract C
+			/// @dev This should appear in Contract C dev doc
+		    event SameSignatureEvent(uint16);
+		    /// @notice This event is defined in Contract C
+			/// @dev This should appear in contract C dev doc
+		    event ContractEvent(uint32);
+		    function f() public {
+		        emit L.SameSignatureEvent(0);
+		        emit SameSignatureEvent(1);
+		        emit L.LibraryEvent(2);
+		        emit ContractEvent(3);
+		    }
+		}
+	)";
+
+	char const* devDoc = R"ABCDEF(
+	{
+	    "events":
+	    {
+	        "ContractEvent(uint32)":
+	        {
+	            "details": "This should appear in contract C dev doc"
+	        },
+	        "SameSignatureEvent(uint16)":
+	        {
+	            "details": "This should appear in Contract C dev doc"
+	        }
+	    },
+	    "kind": "dev",
+	    "methods": {},
+	    "version": 1
+	}
+	)ABCDEF";
+	checkNatspec(sourceCode, "C", devDoc, false);
+
+	char const* userDoc = R"ABCDEF(
+	{
+		"events":
+		{
+			"ContractEvent(uint32)":
+			{
+				"notice": "This event is defined in Contract C"
+			},
+			"LibraryEvent(uint32)":
+			{
+				"notice": "This event is defined in Library L"
+			},
+			"SameSignatureEvent(uint16)":
+			{
+				"notice": "This event is defined in Contract C"
+			}
+		},
+		"kind": "user",
+		"methods": {},
+		"version": 1
+	}
+	)ABCDEF";
+	checkNatspec(sourceCode, "C", userDoc, true);
+}
+
+BOOST_AUTO_TEST_CASE(emit_same_signature_event_different_libraries)
+{
+	char const* sourceCode = R"(
+		library L1 {
+		    /// @notice This event is defined in Library L1
+		    /// @dev This should not appear in Contract C dev doc
+		    event SameSignatureEvent(uint16);
+		}
+		library L2 {
+		    /// @notice This event is defined in Library L2
+		    /// @dev This should not appear in Contract C dev doc
+		    event SameSignatureEvent(uint16);
+		}
```

Comment:
The event should be omitted from both devdoc and userdoc of contract `C`.

Can you test that the event is still included in the userdoc and devdoc of `L1` and `L2`? You should be able to do this by passing library name to `checkNatspec()` instead of `"C"`.

Reply:
Yes, I will add those tests.